### PR TITLE
Get default branch from deployment event

### PR DESCRIPTION
**Pull request by rogermparent**
This PR utilizes the `default_branch` given to us by GitHub to populate the `branch` (previously `main`) field when getting diffs.

The ability to override the branch is kept because it was easy and could be useful in some cases, but in most cases users won't need it.

Test branch at https://github.com/iterative/cml.dev/pull/287

### CODE REVIEW

File: .github/workflows/link-check-deployment-status.yml [hunk 1]
```diff
@@ -2,11 +2,11 @@ name: Check new links against deployment
...
   workflow_call:
     inputs:
-      main:
+      branch:
         type: "string"
         required: false
-        description: "The main branch of this repository"
-        default: "main"
+        description: "The branch to compare against when finding new links to check"
+        default: "${{ github.event.repository.default_branch }}"
```
Comment (casperdcl) [inline .github/workflows/link-check-deployment-status.yml:9]: according to https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability this should actually work fine :)

**Review by casperdcl — APPROVED** [review]
lgtm

**Review by yathomasi — APPROVED** [review]
Looks good to me.

File: .github/workflows/link-check-deployment-status.yml (hunk 1 above)
Comment (rogermparent) [reply .github/workflows/link-check-deployment-status.yml:9]: I was skeptical at first as well and was ready to just axe the option, but I made sure to test this out first and it surprisingly went just fine.

**Pull request merged**